launcher/MangoHud.cpp

```diff
@@ -106,4 +116,37 @@ QString getLibraryString()
 
     return QString();
 }
+
+QString findLibrary(QString libName)
+{
+#ifdef __GLIBC__
+    const char* library = libName.toLocal8Bit().constData();
+
+    void* handle = dlopen(library, RTLD_NOW);
+    if (!handle) {
+        qCritical() << "dlopen() failed:" << dlerror();
+        return {};
+    }
+
+    char path[PATH_MAX];
+    if (dlinfo(handle, RTLD_DI_ORIGIN, path) == -1) {
+        qCritical() << "dlinfo() failed:" << dlerror();
+        dlclose(handle);
+        return {};
+    }
+
+    auto fullPath = FS::PathCombine(QString(path), libName);
+
+    dlclose(handle);
+    return fullPath;
+#else
+    qWarning() << "MangoHud::findLibrary is not implemented on this platform";
+    return {};
+#endif
+}
```

Comment:
:mango:
I'm pretty sure this should be in a different file.

Reply:
It was noted by @Scrumplex that it needed to be changed at some point, and that it would be handled later IIRC

Reply:
yep. I am already using this file in #1160. I think we can just rename it to something like NativeUtils eventually

Reply:
`NativeHelper` or `NativeUtil(s)` would be good, yes